don't panic on 4byte IP addresses

Stebalien opened the pull request:
They're valid.

Stebalien committed:
don't panic on 4byte IP addresses

They're valid.

Stebalien committed:
don't use ipv4 when a zone is set

vyzo commented on net/net.go:
```diff
@@ -115,22 +115,26 @@ func NetAddrToSockaddr(addr net.Addr) Sockaddr {
-	case ip.To4() != nil:
+	if zone != "" {
+		var buf [16]byte
+		copy(buf[:], ip.To16())
```
So, if `ip.To16` returns nil, we'll end up with the unspecific address. 
Is this desirable? Or is the assumption that the user wouldn't pass a zone unless they had a proper ipv6 address?

Stebalien replied:
So, there are two cases her:

1. The IP is nil. In that case, we probably *would* want to default to the unspecified address (icky but go does this as well).
2. The IP is not nil but not a valid ipv6 address (the wrong size, that's the only case). In this case, we should probably return nil.

Stebalien replied:
Fixed. Thoughts?

vyzo replied:
yeah, that's sane behaviour.

Stebalien committed:
cleaner ip to sockaddr conversion

1. Never panic.
2. Only return a unspecified address if the IP is nil (no other random error
cases).
3. Simplify the logic.

vyzo reviewed (approved):
LGTM
